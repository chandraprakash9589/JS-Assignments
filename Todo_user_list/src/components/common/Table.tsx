import React from 'react';

interface TableProps {
  data: any[];
  columns: string[];
}

class Table extends React.Component<TableProps> {
  render() {
    const { data, columns } = this.props;

    return (
      <table>
        <thead>
          <tr>
            {columns.map((column) => (
              <th key={column}>{column}</th>
            ))}
          </tr>
        </thead>
        <tbody>
          {data.map((row, index) => (
            <tr key={index}>
              {columns.map((column) => (
                <td key={column}>{row[column]}</td>
              ))}
            </tr>
          ))}
        </tbody>
      </table>
    );
  }
}

export default Table;
