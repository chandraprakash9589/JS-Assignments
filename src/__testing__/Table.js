import React from "react";
 export default function Table(){
    return(
        <table className="custom-border">
            <thead>
                <tr>
                    <td>Names</td>
                    <td>Email</td>
                </tr>
            </thead>
            <tbody>
            <tr>
                    <td>Demo</td>
                    <td>NA</td>
                </tr>
            </tbody>
        </table>
    )
}