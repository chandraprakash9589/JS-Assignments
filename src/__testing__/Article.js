import React, { Component } from "react";
import Table from "./Table";

export default class Article extends Component {
  constructor(props) {
    super(props);
    this.state = {
      count: 0,
    };
  }
  render() {
    const { count } = this.state;
    return (
      <div>
       
        <div className={`cliks-${count}`}>{count} clicks</div>
        <button
          onClick={() => {
            this.setState({ count: count + 1 });
          }}
        >
          increment
        </button>

        <h1>Article Details</h1>
        <Table />
      </div>
    );
  }
}
