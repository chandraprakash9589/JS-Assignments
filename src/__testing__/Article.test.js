import {shallow,mount} from 'enzyme';
import Article from './Article';
import Table from './Table';
//simulate and drive 
describe("Article page",()=>{
    it("click count",()=>{
        let wrapper = shallow(<Article/>);
        expect(wrapper.find('.clicks-0').length).toEqual(0);
         wrapper.find('button').simulate("click");
        expect(wrapper.find('.click-1').length).toEqual(1);
    })
it("dive", () => {
    let wrapper = shallow(<Article />);
    expect(wrapper.find(Table).dive().find('.custom-border').length).toBe(1);
  });
  it("snapshot 1",()=>{
    let wrapper = shallow(<Table/>);
    expect(wrapper).toMatchSnapshot();
  })
  it("snapshot 2",()=>{
    let wrapper = shallow(<Table/>);
    expect(wrapper).toMatchSnapshot();
  })
  it("snapshot 3",()=>{
    let wrapper = shallow(<Table/>);
    expect(wrapper).toMatchSnapshot();
  })
})

describe.skip("Mount vs shallow",()=>{
    it("snapshot 1",()=>{
        let wrapper = mount(<Table/>);
        console.log(wrapper.debug());
        
      })
})
//Shallow: create and test component only (no children)
// Mount: same as shallow but mounts with children and parent/host component, 
// allows lifecycle methods
// Render: outputs the html given by the component,
//  including children

