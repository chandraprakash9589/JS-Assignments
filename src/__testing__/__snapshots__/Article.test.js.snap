// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article page snapshot  1`] = `
<table
  className="custom-border"
>
  <thead>
    <tr>
      <td>
        Names
      </td>
      <td>
        Email
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Demo
      </td>
      <td>
        NA
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Article page snapshot 1 1`] = `
<table
  className="custom-border"
>
  <thead>
    <tr>
      <td>
        Names
      </td>
      <td>
        Email
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Demo
      </td>
      <td>
        NA
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Article page snapshot 1 2`] = `
<table
  className="custom-border"
>
  <thead>
    <tr>
      <td>
        Names
      </td>
      <td>
        Email
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Demo
      </td>
      <td>
        NA
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Article page snapshot 1 3`] = `
<table
  className="custom-border"
>
  <thead>
    <tr>
      <td>
        Names
      </td>
      <td>
        Email
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Demo
      </td>
      <td>
        NA
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Article page snapshot 2 1`] = `
<table
  className="custom-border"
>
  <thead>
    <tr>
      <td>
        Names
      </td>
      <td>
        Email
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Demo
      </td>
      <td>
        NA
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Article page snapshot 3 1`] = `
<table
  className="custom-border"
>
  <thead>
    <tr>
      <td>
        Names
      </td>
      <td>
        Email
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Demo
      </td>
      <td>
        NA
      </td>
    </tr>
  </tbody>
</table>
`;
